lib/verifier/post_zephir_verifier.rb

```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
+
+require "zlib"
+require_relative "../verifier"
+require_relative "../derivatives"
+require_relative "../derivative/dollar_dup"
+require_relative "../derivative/catalog"
+require_relative "../derivative/rights"
+
+# Verifies that post_zephir workflow stage did what it was supposed to.
+
+module PostZephirProcessing
+  class PostZephirVerifier < Verifier
+    attr_reader :current_date
+
+    def run_for_date(date:)
+      @current_date = date
+      verify_catalog_archive
+      verify_catalog_prep
+      verify_dollar_dup
+      verify_ingest_bibrecords
+      verify_rights
+    end
+
+    # Frequency: ALL
+    # Files: CATALOG_ARCHIVE/zephir_upd_YYYYMMDD.json.gz
+    #   and potentially CATALOG_ARCHIVE/zephir_full_YYYYMMDD_vufind.json.gz
+    # Contents: ndj file with one catalog record per line
+    # Verify:
+    #   readable
+    #   line count must be the same as input JSON
+    def verify_catalog_archive(date: current_date)
+      zephir_update_path = Derivative::CatalogArchive.new(date: date, full: false).path
+      verify_file(path: zephir_update_path)
+      verify_parseable_ndj(path: zephir_update_path)
+
+      if date.last_of_month?
+        ht_bib_export_derivative_params = {
+          location: :ZEPHIR_DATA,
+          name: "ht_bib_export_full_YYYY-MM-DD.json.gz",
+          date: date
+        }
+        output_path = Derivative::CatalogArchive.new(date: date, full: true).path
+        verify_file(path: output_path)
+        verify_parseable_ndj(path: output_path)
+        output_linecount = gzip_linecount(path: output_path)
+
+        input_path = self.class.dated_derivative(**ht_bib_export_derivative_params)
```

Comment:
This one still needs to be extracted; although it isn't a Derivative as such, the same functionality works here 🤔

Reply:
made and implemented `Derivative::HTBibExport`